shared/actions/profile.js

```diff
@@ -652,16 +635,16 @@ function * generatePgpSaga (): SagaGenerator<any, any> {
     const {shouldStoreKeyOnServer} = finishedAction.payload
 
     // $ForceType
-    const {payload: {response}}: IncomingShouldPush = yield take(generatePgpKeyChan, 'keybase.1.pgpUi.shouldPushPrivate')
+    const {response} = yield takeFromChannelMap(generatePgpKeyChanMap, 'keybase.1.pgpUi.shouldPushPrivate')
```

Comment:
That works, and it is what I had originally.

The helper is better in that it can tell you the channel map doesn't have a channel for `'keybase.1.pgpUi.shouldPushPrivate'` and `take` will only result in an error.

While doing: `take(generatePgpKeyChanMap['keybase.1.pgpUi.shouldPushPrivate'])` could lead to `take(undefined)` then the only error you'll get is that you can't take from undefined.